mla/_models.py

```diff
@@ -549,35 +556,81 @@ def __post_init__(self, source: sources.Source, grl: np.ndarray,
         super().__post_init__(source, grl, background_sin_dec_bins)
 
         # Find the run contian in the background time window
-        start_time, end_time = self.background_time_profile.range
-        if withinwindow:
-            fully_contained = (self._grl['start'] >= start_time
-            ) & (self._grl['stop'] < end_time)
-            start_contained = (self._grl['start'] < start_time
-            ) & (self._grl['stop'] > start_time)
-            background_runs = (fully_contained | start_contained)
-            if not np.any(background_runs):
-                print('ERROR: No runs found in GRL for calculation of '
-                      'background rates!')
-                raise RuntimeError
-            background_grl = self._grl[background_runs]
-        else:
-            fully_contained = (
-                self._grl['start'] >= start_time - background_window
-            ) & (self._grl['stop'] < start_time)
-            start_contained = (
-                self._grl['start'] < start_time - background_window
-            ) & (self._grl['stop'] > start_time - background_window)
-            background_runs = (fully_contained | start_contained)
-            if not np.any(background_runs):
-                print('ERROR: No runs found in GRL for calculation of '
-                      'background rates!')
-                raise RuntimeError
-            background_grl = self._grl[background_runs]
+        start, stop = self.background_time_profile.range
+        return_stop_contained = True
+
+        if not withinwindow:
+            start -= background_window
+            stop = start
+            return_stop_contained = False
+
+        background_run_mask = self._contained_run_mask(
+            start,
+            stop,
+            return_stop_contained=return_stop_contained,
+        )
+
+        if not np.any(background_run_mask):
+            print('ERROR: No runs found in GRL for calculation of '
+                  'background rates!')
+            raise RuntimeError
 
+        background_grl = self._grl[background_run_mask]
         self._n_background = background_grl['events'].sum()
         self._n_background /= background_grl['livetime'].sum()
-        self._n_background *= self.background_time_profile.exposure
+        self._n_background *= self._contained_livetime(
```

Comment:
you overwrite the start and stop here for not withinwindow. It should always be the start and end of the background time profile